static/app/views/issueList/draggableTabBar.tsx

```diff
@@ -24,17 +25,74 @@ export interface Tab {
 
 export interface DraggableTabBarProps {
   tabs: Tab[];
+  onAddView?: React.MouseEventHandler;
+  /**
+   * Callback function to be called when user clicks the `Delete` button.
+   * Note: The `Delete` button only appears for persistent views
+   */
   onDelete?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks on the `Discard Changes` button.
+   * Note: The `Discard Changes` button only appears for persistent views when `isChanged=true`
+   */
   onDiscard?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks on the `Discard` button for temporary views.
+   * Note: The `Discard` button only appears for temporary views
+   */
+  onDiscardTempView?: () => void;
+  /**
+   * Callback function to be called when user clicks the 'Duplicate' button.
+   * Note: The `Duplicate` button only appears for persistent views
+   */
   onDuplicate?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Rename' button.
+   * Note: The `Rename` button only appears for persistent views
+   */
   onRename?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Save' button.
+   * Note: The `Save` button only appears for persistent views when `isChanged=true`
+   */
   onSave?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Save View' button for temporary views.
+   */
+  onSaveTempView?: () => void;
+  showTempTab?: boolean;
+  tempTabContent?: React.ReactNode;
+  tempTabLabel?: string;
 }
 
-export function DraggableTabBar(props: DraggableTabBarProps) {
-  const [tabs, setTabs] = useState<Tab[]>(props.tabs);
+export function DraggableTabBar({
+  showTempTab = false,
+  tempTabContent,
+  tempTabLabel = 'Unsaved',
+  onAddView,
+  onDelete,
+  onDiscard,
+  onDuplicate,
+  onRename,
+  onSave,
+  onDiscardTempView,
+  onSaveTempView,
+  ...props
+}: DraggableTabBarProps) {
+  const tempTab = {
+    key: 'temporary-tab',
+    label: tempTabLabel,
+    content: tempTabContent,
+  };
+  const [tabs, setTabs] = useState<Tab[]>([...props.tabs, tempTab]);
   const [selectedTabKey, setSelectedTabKey] = useState<Key>(props.tabs[0].key);
 
+  useEffect(() => {
+    if (!showTempTab && selectedTabKey === 'temporary-tab') {
+      setSelectedTabKey(props.tabs[0].key);
+    }
+  }, [showTempTab, selectedTabKey, props.tabs]);
```

Comment:
If the temp tab goes away, switch to the first tab. (Ideally this would be the most recent tab, but that's out of the scope of this PR)